lib/ufe/UsdStageMap.h

```diff
@@ -22,22 +22,37 @@
 #include <pxr/usd/usd/stage.h>
 #include <pxr/base/tf/hash.h>
 
+#include <maya/MObjectHandle.h>
+
 #include <unordered_map>
 
+// Allow for use of MObjectHandle with std::unordered_map.
+namespace std {
+template <> struct hash<MObjectHandle> {
+    std::size_t operator()(const MObjectHandle& handle) const {
+      return static_cast<std::size_t>(handle.hashCode());
+    }
+};
+}
+
 PXR_NAMESPACE_USING_DIRECTIVE
 
 MAYAUSD_NS_DEF {
 namespace ufe {
 
 //! \brief USD Stage Map
 /*!
-	Map of AL_usdmaya_ProxyShape UFE path to corresponding stage.
-
-	Map of stage to corresponding AL_usdmaya_ProxyShape UFE path.  Ideally, we
-	would support dynamically computing the path for the AL_usdmaya_ProxyShape
-	node, but we assume here it will not be reparented.  We will also assume that
-	a USD stage will not be instanced (even though nothing in the data model
-	prevents it).
+	Two-way map of proxy shape UFE path to corresponding stage.
+
+	We will assume that	a USD proxy shape will not be instanced (even though
```

Comment:
Previous USD stage map was path-based, which was impossible to maintain in the presence of path changes through renaming.  Simply use the proxy shape MObject instead.

Reply:
Again, slightly concerned about the no instancing assumption, but maybe I'm misunderstanding.

Reply:
No, you're not misunderstanding...  As per other comment entered as
https://github.com/Autodesk/maya-usd/issues/256